lib/client.js

```diff
@@ -120,6 +125,27 @@ function* createClient(url, options) {
   function publish(exchange, rootingKey, message, opts) {
     return busClient.channel.publish(exchange, rootingKey, new Buffer(JSON.stringify(message)), opts);
   }
+
+  function* _connect() {
+    busClient.connection = yield amqplib.connect(url, options);
+    busClient.channel = yield busClient.connection.createChannel();
+    // see error handling and reconnection https://www.cloudamqp.com/docs/nodejs.html
+    // http://www.squaremobius.net/amqp.node/channel_api.html#failure
+    busClient.connection.on('error', _onDisconnection);
+    busClient.emit('connected');
+  }
+
+  /**
+   * @return {void}
+   */
+  function _onDisconnection(err) {
+    // for some reason the on('error') event is called twice
```

Comment:
Are you sure about that since there is a call that should not be here in the constructor ?